src/api-client.js

```diff
@@ -0,0 +1,94 @@
+const requestOptions = ({
+  method,
+  content_type = "application/json",
+  token = null,
+  body = null,
+}) => {
+  const headers = { "Content-Type": content_type };
+  if (token !== null) {
+    headers["Authorization"] = `Bearer ${token}`;
+  }
+  if (method !== "GET" && method !== "DELETE") {
+    body = JSON.stringify(body);
+    return {
+      method: method,
+      headers: headers,
+      body: body,
+    };
+  }
+
+  return {
+    method: method,
+    headers: headers,
+  };
+};
+
+export class ApiClient {
+  static getTitle = async () => {
+    return await fetch("/api", requestOptions({ method: "GET" }));
```

Comment:
I would recommend to look at Axios - usefull helper for http client - https://axios-http.com/docs/api_intro